File: files/CH-186.R
library(tidyverse)
library(readxl)

path = "files/CH-186 Column Splitting.xlsx"
input = read_excel(path, range = "B2:B8")
test  = read_excel(path, range = "D2:E8") %>%
  mutate_all(~replace_na(., ""))

result = input %>%
  mutate(ID.1 = str_remove_all(ID, "[^aeiouAEIOU]"),
         ID.2 = str_remove_all(ID, "[aeiouAEIOU]")) %>%
  select(-ID)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE         

File: files/CH-186.py
import pandas as pd
import numpy as np

path = "CH-186 Column Splitting.xlsx"

input = pd.read_excel(path, usecols="B", skiprows=1, nrows=7)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=7).fillna("").astype('str')

input['ID.1'] = input['ID'].str.replace(r'[^aeiouAEIOU]', '', regex=True)
input['ID.2'] = input['ID'].str.replace(r'[aeiouAEIOU]', '', regex=True)
result = input.drop(columns=['ID'])

print(result.equals(test)) # True